### PR TITLE
Added token generation

### PR DESCRIPTION
**Pull request by assertchris**

### CODE REVIEW

File: code/models/ShareToken.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+		'Token' => 'Varchar(16)',
+	);
+
+	/**
+	 * @var array
+	 */
+	private static $has_one = array(
+		'Page' => 'Page'
+	);
+}
```
Comment (tractorcow) [inline code/models/ShareToken.php]: No newline at end of file make homer something something.

File: code/extensions/ShareDraftContentExtension.php [hunk 2]
```diff
@@ -1,5 +1,78 @@
...
+	 */
+	protected function getNewShareToken() {
+		if(!$this->owner->ShareTokenSalt) {
+			$this->owner->ShareTokenSalt = $this->getNewToken();
+			$this->owner->write();
+		}
+
+		$token = null;
+
+		while(true) {
```
Comment (tractorcow) [inline code/extensions/ShareDraftContentExtension.php]: I'm scared. Can you have some kind of limit on this?

I also notice your first iteration of this loop tests `"Token" => null`. Is this intentional?
